packages/query/src/block-processor.ts

```diff
@@ -153,24 +162,16 @@ export class BlockProcessor implements BlockProcessorInterface {
       { retry: () => true },
     );
 
-    // TODO: init validator info in a better way, possibly after batch endpoint
-    // implemented https://github.com/penumbra-zone/penumbra/issues/4688
+    // special case genesis sync
     if (currentHeight === -1n) {
-      // In the `for` loop below, we only update validator infos once we've
-      // reached the latest known epoch. This means that, if a user is syncing
-      // for the first time, they could experience a broken UI until the latest
-      // known epoch is reached, since they may have delegation tokens but no
-      // validator info to go with them. So we'll update validator infos at the
-      // beginning of sync as well, and force the rest of sync to wait until
-      // it's done.
-      void this.updateValidatorInfos(0n);
-    }
-
-    // if this is the first block being synced and the bundled genesis block data is available,
-    // process the genesis block first
-    if (currentHeight === -1n && this.genesisBlock) {
-      await this.processBlock(this.genesisBlock, latestKnownBlockHeight);
-      currentHeight = 0n;
+      if (this.genesisBlock?.height === currentHeight + 1n) {
+        // begin the chain with bundled genesis block if provided
+        await this.processBlock(this.genesisBlock, latestKnownBlockHeight);
+        currentHeight = this.genesisBlock.height;
```

Comment:
i believe these should be transposed (height first, then process) as that's the technique used in the loop

https://github.com/penumbra-zone/web/commit/8fe9234c988abf4b284b3996da98963ec430b631#r145572205

Reply:
note from offline conversation; if processing the genesis fails, and we update the current height before that, what happens in the the exponential backoff? [this.viewServer.resetTreeToStored()](https://github.com/penumbra-zone/web/blob/8fe9234c988abf4b284b3996da98963ec430b631/packages/query/src/block-processor.ts#L124) re-queries idb for local sync height, effectively preventing this class of issues from occurring.